lib/src/components/NxToast/NxToastProvider.tsx

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+
+import React, { useCallback, useState } from 'react';
+import ToastContext from './contexts';
+import NxToastContainer from './NxToastContainer';
+import { NxToastProviderProps, nxToastProviderPropTypes, ToastAddModel, ToastModel } from './types';
+
+let id = 1;
+
+const NxToastProvider = (props: NxToastProviderProps) => {
+  const { children } = props;
+  const [toasts, setToasts] = useState<ToastModel[]>([]);
+  const [activeElementNotToast, setActiveElementNotToast] = useState<HTMLElement | null>(null);
```

Comment:
I believe this one can be a ref.  When it really comes down to it, the only difference between a ref and a piece of state is that the piece of state triggers a re-render when its value is changed.  For items where we don't need or want that, a ref is better.